modules/age.nix

```
@@ -104,7 +125,13 @@ let
         '';
       };
       file = mkOption {
-        type = types.path;
+        type = mkOptionType {
+          name = "nix-path";
+          descriptionClass = "noun";
+          check = builtins.isPath;
+          merge = mergeEqualOption;
+        };
+
```

```suggestion
        type = types.package;
```
Does this work? See https://github.com/NixOS/nixpkgs/blob/master/lib/types.nix#L464-L473

---

No, because at the moment the value is typechecked, it's not yet a store path but a mere path.

---

I see, yes. `builtins.toString /my/repo/secrets/foo.nix` yields `"/my/repo/secrets/foo.nix"` not the store path where it'll get stored.